src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -200,10 +208,59 @@ internal int SerializeLogRecord(LogRecord logRecord)
             var name = logRecord.CategoryName;
 
             // If user configured explicit TableName, use it.
-            if (this.m_tableMappings == null || !this.m_tableMappings.TryGetValue(name, out var eventName))
+            if (this.m_tableMappings == null || (!this.m_tableMappings.TryGetValue(name, out var eventName) && !this.shouldPassThruTableMappings))
             {
                 eventName = this.m_defaultEventName;
             }
+            else if (this.shouldPassThruTableMappings && eventName == null)
+            {
+                char[] tempArr = new char[name.Length];
```

Comment:
If the max length allowed is 32, we only need a temporary array of size 32.

I'm curious why this is not counting towards allocation in the benchmark results.

Reply:
Okay it looks like the results that you have pasted in the PR description show no extra allocation but the one in the benchmark test file show the extra allocation:

|                                          Method |     Mean |     Error |    StdDev |   Median |  Gen 0 | Allocated |
|------------------------------------------------ |---------:|----------:|----------:|---------:|-------:|----------:|
| CategoryTableNameMappingsDefinedInConfiguration | 1.873 us | 0.0442 us | 0.1218 us | 1.839 us | 0.0610 |     256 B |
|    PassThruTableNameMappingsWhenTheRuleIsEnbled | 1.824 us | 0.0358 us | 0.0317 us | 1.817 us | 0.0877 |     **368 B** |

Which is the correct one? I believe it's the one in the test class file. You might want to have a `Threadlocal<char[]>` with the size 32 to avoid allocation in the hot path.